receiver/awscontainerinsightreceiver/internal/k8swindows/kubelet.go

```diff
@@ -0,0 +1,88 @@
+package k8swindows
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	ci "github.com/open-telemetry/opentelemetry-collector-contrib/internal/aws/containerinsight"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/cadvisor/extractors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/host"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awscontainerinsightreceiver/internal/stores/kubeletutil"
+
+	"go.uber.org/zap"
+	stats "k8s.io/kubelet/pkg/apis/stats/v1alpha1"
+)
+
+type kubeletSummaryProvider struct {
+	logger   *zap.Logger
+	hostIP   string
+	hostPort string
+	client   *kubeletutil.KubeletClient
+	hostInfo host.Info
+}
+
+func new(logger *zap.Logger, info host.Info) (*kubeletSummaryProvider, error) {
+	hostIP := os.Getenv("HOST_IP")
+	kclient, err := kubeletutil.NewKubeletClient(hostIP, ci.KubeSecurePort, logger)
+	if err != nil {
+		return nil, errors.New("failed to initialize kubelet client")
+	}
+	return &kubeletSummaryProvider{
+		logger:   logger,
+		client:   kclient,
+		hostInfo: info,
+	}, nil
+}
+
+func (k *kubeletSummaryProvider) getMetrics() ([]*extractors.CAdvisorMetric, error) {
+	summary, err := k.client.Summary(k.logger)
+	if err != nil {
+		k.logger.Error("kubelet summary API failed, ", zap.Error(err))
+		return nil, err
+	}
+
+	return k.getPodMetrics(summary)
+}
+
+func (k *kubeletSummaryProvider) getContainerMetrics(summary *stats.Summary) ([]*extractors.CAdvisorMetric, error) {
+	var metrics []*extractors.CAdvisorMetric
+	// todo: implement CPU, memory metrics from containers
+	return metrics, nil
+}
+
+func (k *kubeletSummaryProvider) getPodMetrics(summary *stats.Summary) ([]*extractors.CAdvisorMetric, error) {
+	// todo: This is not complete implementation of pod level metric collection since network level metrics are pending
+	// May need to add some more pod level labels for store decorators to work properly
+
+	var metrics []*extractors.CAdvisorMetric
+
+	nodeCPUCores := k.hostInfo.GetNumCores()
+	for _, pod := range summary.Pods {
+		k.logger.Info(fmt.Sprintf("pod summary %v", pod.PodRef.Name))
```

Comment:
Nit: Is there any way we can also show how many containers are running in the pod? If yes, it might be a nice to have. If not then nevermind.

Reply:
This information will be collected in separate metrics field but not exactly the number of containers rather cpu, memory metrics for each container in pod.